aten/src/ATen/native/Loss.cpp

```diff
@@ -30,13 +30,14 @@ DEFINE_DISPATCH(mse_stub);
 DEFINE_DISPATCH(mse_backward_stub);
 
 Tensor cosine_embedding_loss(const Tensor& input1, const Tensor& input2, const Tensor& target, double margin, int64_t reduction) {
+  auto targ_dim = target.dim();
   TORCH_CHECK(
-      target.dim() == 1,
-      "1D target tensor expected, multi-target not supported");
+      targ_dim == 1 || targ_dim == 0,
+      "0D or 1D target tensor expected, multi-target not supported");
```

Comment:
I think this check should be expanded a bit to ensure that the input and target shapes match.

Currently, it's possible to pass:
* (N, D) inputs with () target - does something silently wrong
* (D) inputs with (N) target - produces an unclear error message